Toggl.Foundation.MvvmCross/ViewModels/ForgotPasswordViewModel.cs

```
@@ -17,75 +14,98 @@
 
 namespace Toggl.Foundation.MvvmCross.ViewModels
 {
+    public struct PasswordResetResult
```

I don't understand why do we need this. You could as easily leave it as it is and use the `Errors` property of the `Action` to show the error message, basically the way it was. I think it makes more sense semantically (errors are returned through the `Errors` property). Can you justify the change?

---

You are partially right. I need to introduce this `Result` struct because we need to return a custom error message for the exceptions which is handled by the VM's `private string toErrorString(Exception exception)`

---

I still don't understand. The error message was already being passed to the VC in  `public IObservable<string> ErrorMessage { get; }`